core/src/test/scala/unit/kafka/log/LogCleanerIntegrationTest.scala

```diff
@@ -185,4 +185,20 @@ class LogCleanerIntegrationTest extends AbstractLogCleanerIntegrationTest {
       (key, curValue)
     }
   }
+
+  @Test
+  def testDeadThreadCountMetricOnUnexpectedError(): Unit = {
```

Comment:
This is kind of a nitpick, but we're not actually testing the metric here. We're testing the source of it. Perhaps we should also have a test case which checks the metric directly. This helps us avoid name compatibility problems if there is a class renaming.

Reply:
Ok, I'll add an assert for that.

Reply:
Couldn't really make a separate case easily as the metrics registry is a singleton object, it is cumbersome to hack around the static initialization/shutdown.